### PR TITLE
EOS-20358: Custom table component with filter and pagination

### PR DESCRIPTION
**Pull request by NareshGopal**
Signed-off-by: Naresh Gopalakrishnan <naresh.gopalakrishnan@seagate.com>

## Problem Statement
<pre>
<code>
EOS-20358:CSM-UI: Filter components for table
</code>
</pre>
## Unit testing on RPM done
<pre>
<code>
No
</code>
</pre>
## Problem Description
<pre>
<code>
A custom table component has to be built. It should contain filter component at the top and pagination component at the bottom.

</code>
</pre>
## Solution/Screenshots
 A custom table component named 'cortx-v-data-table' has been created which uses Vuetify's 'v-data-table' component underneath. Improvement will be done on this component. 

Top of this table contains a filter component which can be used to filter records from the entire table or from certain columns  based on the search string. Since back-end for filtering is not complete yet, filtering won't function as of now. But the calls are made to the back-end upon triggering the filter. 

UI of the filter component 

image

Calls made to back-end with query parameters "timestamp=test&user=test" can be seen in the below screenshot. So this would work eventually when the back-end is ready.

image

Vuetify's 'v-pagination' component was used for building custom pagination component.

image

**IMPORTANT NOTE: Custom component is not complete yet. It cannot take slots and hence it cannot be used everywhere in the project. This component will be improved continuously so that it can be a component of CORTX UI library.**

## Unit Test Cases
<pre>
<code>
1. Checked the proper render of the table component
2. Checked whether proper calls are made to back-end for filtering component.
3. Checked whether pagination functionality and 'rows per page' functionality works properly.
</code>
</pre>

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

**Review by kupranay — APPROVED** [review]

File: gui/src/components/maintenance/cortx-auditlog.vue [hunk 1]
```diff
@@ -15,7 +15,7 @@
 * please email opensource@seagate.com or cortx-questions@seagate.com.
 */
 <template>
-  <div v-feature="unsupportedFeatures.auditlog">
```
Comment (shrimetta) [inline gui/src/components/maintenance/cortx-auditlog.vue:18]: Why is this removed? This should not be removed. The decision will be in backend whether auditlog feature is supported or not.

File: gui/src/components/maintenance/cortx-auditlog.vue [hunk 2]
```diff
@@ -70,79 +70,23 @@
...
-            id="auditlog-data"
-          >{{ log }}</span>
+          <cortx-data-table
+            :headers="auditLogTableHeaderList" 
+            :records="auditLog.logs" 
+            :onSort="onAuditLogSort"
+            :onFilter="onAuditLogFilter" 
+            :sortParams="auditLogQueryParams"
+            :rowsPerPage="[10, 20, 30, 50, 100, 150, 200]" 
+            class="cortx-table"
```
Comment (shrimetta) [inline gui/src/components/maintenance/cortx-auditlog.vue:85]: Is this still required as styling is now moved to cortx-data-table?

File: gui/src/components/maintenance/cortx-auditlog.vue [hunk 3]
```diff
@@ -294,12 +197,43 @@ export default class CortxAuditLog extends Vue {
...
 
+  public async clearFilters() {
+    for (const header of this.auditLogTableHeaderList) {
+      delete this.auditLogQueryParams[header.field_id]
+    }
+  }
+
+  public async onAuditLogFilter(headerFields: string[], value: string) {
+    if(value.length > 0) {
+      console.log("inside on audit log filter")
```
Comment (shrimetta) [inline gui/src/components/maintenance/cortx-auditlog.vue:208]: Please remove console logging.

**Review by shrimetta — APPROVED** [review]
